bug: v10 @trpc/client has undefined `global` in workers

### Provide environment information

```
  System:
    OS: macOS 12.5
    CPU: (8) arm64 Apple M1
    Memory: 158.41 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.5.0 - ~/.nvm/versions/node/v18.5.0/bin/node
    Yarn: 1.22.17 - ~/.yarn/bin/yarn
    npm: 8.12.1 - ~/.nvm/versions/node/v18.5.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Chrome Canary: 106.0.5249.0
    Firefox Developer Edition: 104.0
    Safari: 15.6
  npmPackages:
    @trpc/client: 10.0.0-alpha.47 => 10.0.0-alpha.47 
    @trpc/server: =10.0.0-alpha.47 => 10.0.0-alpha.47 
```

### Describe the bug

When attempting to deploy my app I was getting the following error:
```
✘ [ERROR] A request to the Cloudflare API (/accounts/000000011111111122222222/workers/scripts/app-edge) failed.

  Uncaught ReferenceError: global is not defined
    at worker.js:10:10
   [code: 10021]

  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose
```

### To reproduce

Try to deploy something using something akin to my gist for [wrapping Durable Objects in TRPC](https://gist.github.com/colelawrence/ba912db3cd1539ee820c3fe5e67654df)

### Additional information

I was able to fix in the short term by adding a `banner` via esbuild: `if (typeof global === "undefined") { globalThis.global = globalThis; }`

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

--- a/packages/client/src/getFetch.ts
+++ b/packages/client/src/getFetch.ts
@@ -4,11 +4,13 @@ export function getFetch(f?: typeof fetch): typeof fetch {
   if (f) {
     return f;
   }
+
   const win = getWindow();
-  if (win.fetch) {
-    return typeof win.fetch.bind === 'function'
-      ? win.fetch.bind(win)
-      : win.fetch;
+  const globalFetch = win.fetch;
+  if (globalFetch) {
+    return typeof globalFetch.bind === 'function'
+      ? globalFetch.bind(win)
+      : globalFetch;
   }
 
   throw new Error('No fetch implementation found');

--- a/packages/client/src/internals/fetchHelpers.ts
+++ b/packages/client/src/internals/fetchHelpers.ts
@@ -2,11 +2,10 @@ export function getWindow() {
   if (typeof window !== 'undefined') {
     return window;
   }
-  return global;
+  return globalThis;
 }
 export function getAbortController(
   ac?: typeof AbortController,
 ): typeof AbortController | undefined {
-  const win = getWindow();
-  return ac ?? win.AbortController ?? undefined;
+  return ac ?? getWindow().AbortController ?? undefined;
 }

